apps/src/lib/kits/maker/ui/SetupChecklist.jsx

```diff
@@ -69,7 +71,13 @@ export default class SetupChecklist extends Component {
   }
 
   detect() {
-    const {setupChecker} = this.props;
+    const {webSerialPort} = this.props;
+    let {setupChecker} = this.props;
+    // If a webSerialPort has been provided, create a new SetupChecker with the port each detection cycle.
+    if (webSerialPort) {
+      const wrappedSerialPort = new WebSerialPortWrapper(webSerialPort);
+      setupChecker = new SetupChecker(wrappedSerialPort);
```

Comment:
props are readonly ([react docs](https://reactjs.org/docs/components-and-props.html#props-are-read-only)). if a child needs to trigger a prop update, usually 2 things are true:
1. that prop should be part of the parent's state
2. that state needs and update handler for the child to call to trigger that state update, which will then propagate down to the child as props again